corehq/apps/app_manager/exceptions.py

```diff
@@ -214,28 +215,75 @@ class AppInDifferentDomainException(AppManagerException):
     pass
 
 
-class InvalidPropertyException(Exception):
+class FormActionsDiffException(Exception):
+    """
+    A parent exception class to handle all errors that occur while applying diffs
+    """
+    def get_user_message(self):
+        return _("Form Actions Diff failed to be applied successfully")
+
+
+class InvalidPropertyException(FormActionsDiffException):
+    FORMAT_STRING = "Invalid key found: {}"
+
     def __init__(self, invalid_property):
         self.invalid_property = invalid_property
-        message = f"Invalid key found: {self.invalid_property}"
+        message = self.FORMAT_STRING.format(self.invalid_property)
         super().__init__(message)
 
+    def get_user_message(self):
+        return _(self.FORMAT_STRING).format(self.invalid_property)
 
-class MissingPropertyException(Exception):
-    def __init__(self, *missing_properties):
-        self.missing_properties = missing_properties
-        if self.missing_properties:
-            message = f"The following properties were not found: {', '.join(self.missing_properties)}"
-        else:
-            message = "No missing properties specified"
+
+class MissingPropertyMapException(FormActionsDiffException):
```

Comment:
Can you give me an example that will result in `MissingPropertyMapException`? I have a hard time connectting the code logic to the actual user actions.

Reply:
Sure, the example you're looking for is [this test case](https://github.com/dimagi/commcare-hq/blob/mjr/show_multi_updates/corehq/apps/app_manager/tests/test_models.py#L516-L525). If you try to update a property that someone else deleted, `MissingPropertyMapException` will be raised